tools/build_defs/repo/http.bzl

```diff
@@ -38,103 +38,35 @@ replace the native rules.
 """
 
 load(
-    ":utils.bzl",
+    "//tools/build_defs/repo/private:utils.bzl",
```

Comment:
also I think this is a breaking change -- some people have been using stuff from utils.bzl, IIRC

Reply:
I added forwarding, but I agree that this turned out to be more intrusive than I thought, mostly because the rule has a lot of transitive symbol deps. I will look into moving out only the `Label` part.

Reply:
I moved this into https://github.com/bazelbuild/bazel/pull/19997, where it fits much better. This PR can be reviewed and merged independently.